src/main/java/org/sagebionetworks/template/global/GlobalResourcesBuilderImpl.java

```diff
@@ -0,0 +1,93 @@
+import org.sagebionetworks.template.LoggerFactory;
+import org.sagebionetworks.template.SesClient;
+import org.sagebionetworks.template.StackTagsProvider;
+import org.sagebionetworks.template.config.Configuration;
+
+import java.io.StringWriter;
+
+import static org.sagebionetworks.template.Constants.GLOBAL_RESOURCES_STACK_NAME_FORMAT;
+import static org.sagebionetworks.template.Constants.JSON_INDENT;
+import static org.sagebionetworks.template.Constants.PROPERTY_KEY_STACK;
+import static org.sagebionetworks.template.Constants.SES_SYNAPSE_DOMAIN;
+import static org.sagebionetworks.template.Constants.STACK;
+import static org.sagebionetworks.template.Constants.TEMPLATE_GLOBAL_RESOURCES_SNS;
+import static org.sagebionetworks.template.Constants.GLOBAL_CFSTACK_OUTPUT_KEY_SES_BOUNCE_TOPIC;
+import static org.sagebionetworks.template.Constants.GLOBAL_CFSTACK_OUTPUT_KEY_SES_COMPLAINT_TOPIC;
+
+public class GlobalResourcesBuilderImpl implements GlobalResourcesBuilder {
+
+    CloudFormationClient cloudFormationClient;
+    VelocityEngine velocityEngine;
+    Configuration config;
+    Logger logger;
+    StackTagsProvider stackTagsProvider;
+    SesClient sesClient;
+
+    @Inject
+    public GlobalResourcesBuilderImpl(CloudFormationClient cloudFormationClient,
+                                      VelocityEngine velocityEngine,
+                                      Configuration config,
+                                      LoggerFactory loggerFactory,
+                                      StackTagsProvider stackTagsProvider,
+                                      SesClient sesClient) {
+        this.cloudFormationClient = cloudFormationClient;
+        this.velocityEngine = velocityEngine;
+        this.config = config;
+        this.logger = loggerFactory.getLogger(GlobalResourcesBuilderImpl.class);
+        this.stackTagsProvider = stackTagsProvider;
+        this.sesClient = sesClient;
+    }
+
+    @Override
+    public void buildGlobalResources() throws InterruptedException {
+        String stackName = createStackName();
+        VelocityContext context = createContext();
+        Template template = velocityEngine.getTemplate(TEMPLATE_GLOBAL_RESOURCES_SNS);
+        StringWriter stringWriter = new StringWriter();
+        template.merge(context, stringWriter);
+        String resultJSON = stringWriter.toString();
+        JSONObject templateJson = new JSONObject(resultJSON);
+        resultJSON = templateJson.toString(JSON_INDENT);
+        //this.logger.info(resultJSON);
+        //System.out.println(resultJSON);
+        cloudFormationClient.createOrUpdateStack(new CreateOrUpdateStackRequest()
+            .withStackName(stackName)
+            .withTemplateBody(resultJSON)
+            .withTags(stackTagsProvider.getStackTags())
+        );
+        cloudFormationClient.waitForStackToComplete(stackName);
+        // setup SES notifications on prod stack
+        if (config.getProperty(PROPERTY_KEY_STACK) == "prod") {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.